queries/dockerfile/injections.scm

```diff
@@ -1,10 +1,9 @@
 ((comment) @injection.content
   (#set! injection.language "comment"))
 
-((shell_command
-  (shell_fragment) @injection.content)
+((shell_command) @injection.content
```

Comment:
There's a reason it was implemented the way it was.

This includes the dockerfile heredoc in the bash injection.
```dockerfile
RUN <<EOF
echo 'hello world'
EOF

Reply:
But `heredoc_marker` is also under `shell_fragment` anyway, right now at least. So it doesn't seem it will change much in that regard. Or am I missing something?

On the other hand highlighting individual `shell_fragment` nodes results in a broken highlighting for snippets like this one:

```dockerfile
RUN ls \
  --tree
```

Here each line results in a separate `shell_fragment`, which results in `--tree` being treated as a separate `command` instead of being an `argument`. Not a big deal probably, but not exactly correct still.

Reply:
The tree with the change from PR:

```scm
(run_instruction) ; [1:1 - 2:8]
 (shell_command) ; [1:5 - 2:8]
  (command) ; [1:5 - 2:8]
   name: (command_name) ; [1:5 - 6]
    (word) ; [1:5 - 6]
   argument: (word) ; [2:3 - 8]
  (shell_fragment) ; [1:5 - 7]
  (line_continuation) ; [1:8 - 2:0]
  (shell_fragment) ; [2:3 - 8]
```

The tree with original `shell_fragment` matching:

```scm
(run_instruction) ; [1:1 - 2:8]
 (shell_command) ; [1:5 - 2:8]
  (shell_fragment) ; [1:5 - 7]
   (command) ; [1:5 - 6]
    name: (command_name) ; [1:5 - 6]
     (word) ; [1:5 - 6]
  (line_continuation) ; [1:8 - 2:0]
  (shell_fragment) ; [2:3 - 8]
   (command) ; [2:3 - 8]
    name: (command_name) ; [2:3 - 8]
     (word) ; [2:3 - 8]
```

Reply:
Your version will work if `heredoc_marker` is moved.

Reply:
>Your version will work if heredoc_marker is moved.

Sorry, I don't get if it's for or against the change. Should I revert that part anyway?

Reply:
Against, until the parser is fixed.

Reply:
This does not change anything.

Reply:
The start of a heredoc is still captured by it, by the way, so that didn't change

Reply:
But that injection breaks highlighting at the moment and I don't see why using `injection.combined` is necessary here. Treating all of the nodes as a big `bash` script doesn't seem to make sense here. They're meant to be distinct separate scripts. As I understand, `injection.combined` is meant for languages like PHP, where you can stop PHP at basically any point (like at the start of the loop), put a bunch of HTML and the get back to PHP. For dockerfile+bash that doesn't seem to make sense.

Reply:
> But that injection breaks highlighting at the moment

So does the alternative.

> Treating all of the nodes as a big `bash` script doesn't seem to make sense here. They're meant to be distinct separate scripts.

This has multiple nodes but is a single script:

```dockerfile
RUN apt-get update && \
    apt-get install foo
```

---

The **only** solution here is to fix the parser first and *then* use your injection.

Reply:
>So does the alternative.

Not sure what you mean. Can you show an example where an implementation in this PR breaks the highlighting?

>This has multiple nodes but is a single script:

In the case you're showing it makes sense. But multiple `RUN` statements should not be treated as a single giant script.

Reply:
I did. In my very first comment.